### PR TITLE
Support configurable msgstream

### PR DESCRIPTION
**Pull request by xiyichan**
issue:#15674

### CODE REVIEW

**Comment by jaime0815** [comment]
relate to https://github.com/milvus-io/milvus/pull/16110

File: internal/mq/msgstream/mq_factory.go [hunk 1]
```diff
@@ -86,12 +86,10 @@ type RmsFactory struct {
...
 
-// SetParams is used to set parameters for RmsFactory
-func (f *RmsFactory) SetParams(params map[string]interface{}) error {
-	err := mapstructure.Decode(params, f)
-	if err != nil {
-		return err
-	}
+// Init is used to set parameters for RmsFactory
+func (f *RmsFactory) Init(params *paramtable.ComponentParam) error {
+	f.RmqBufSize = 1024
```
Comment (xiaofan-luan) [inline internal/mq/msgstream/mq_factory.go:91]: why rmqbufSize and ReceiveBufSize is not configurable?

File: internal/util/paramtable/component_param.go [hunk 2]
```diff
@@ -75,6 +75,14 @@ func (p *ComponentParam) SetLogConfig(role string) {
 	p.BaseTable.SetLogConfig()
 }
 
+func (p *ComponentParam) RocksmqEnable() bool {
```
Comment (xiaofan-luan) [inline internal/util/paramtable/component_param.go:78]: for standalone we only support rmq.
So I guess rocksmqEnable is not a necessary config.

We need PulsarEnable/KafkaEnable in the distributed mode

**Comment by xiaofan-luan** [comment]
for NewRmsFactory implementation, we load a lot of params inside InitRocksMQ func, which I believe is a temporary hack.

All the parameters should be initiated in mq_factory, we can do it in the next patch.

Anyway the implementation is a big progress, thanks @xiyichan @jaime0815

File: internal/mq/msgstream/mq_factory.go (hunk 1 above)
Comment (xiyichan) [reply internal/mq/msgstream/mq_factory.go:91]: The old code is written as 1024 and this configuration does not exist in milvus.yaml

File: internal/util/paramtable/component_param.go (hunk 2 above)
Comment (xiyichan) [reply internal/util/paramtable/component_param.go:78]: I think it would be better to offer more options.

**Comment by jaime0815** [comment]
> InitRocksMQ

It seems InitRocksMQ belongs to the server side operation,  the client side parameters should be initiated within mq_factory.

File: internal/mq/msgstream/mq_factory.go (hunk 1 above)
Comment (xiaofan-luan) [reply internal/mq/msgstream/mq_factory.go:91]: > d code is written as 1024 and this configuration does not exist in milvus.yaml

Ok, let's leave it to next commit

**Pull request merged**